pkg/api/admission/ownerref/ownerref_admission.go

```diff
@@ -0,0 +1,53 @@
+package ownerref
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api/meta"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+func init() {
+	admission.RegisterPlugin("openshift.io/OwnerReference",
+		func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+			return NewOwnerReferenceBlocker()
+		})
+}
+
+// ownerReferenceAdmission implements an admission controller that stops anyone from setting owner references
+// TODO enforce this check based on the level of access allowed.  If you can delete the object, you can add an
+// owner reference.  We need to make sure the upstream controller works before allowing this.
+type ownerReferenceAdmission struct {
+	*admission.Handler
+}
+
+func NewOwnerReferenceBlocker() (admission.Interface, error) {
+	return &ownerReferenceAdmission{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}, nil
+}
+
+// Admit makes admission decisions while enforcing ownerReference
+func (q *ownerReferenceAdmission) Admit(a admission.Attributes) (err error) {
+	// ignore all operations that correspond to sub-resource actions
+	if len(a.GetSubresource()) != 0 {
+		return nil
+	}
+
+	metadata, err := meta.Accessor(a.GetObject())
+	if err != nil {
+		// if we don't have object meta, we don't have fields we're trying to control
+		return nil
+	}
+
+	if ownerRefs := metadata.GetOwnerReferences(); len(ownerRefs) > 0 {
```

Comment:
> should we only enforce on mutation (instead of just presence) if GetOldObject() is available? otherwise, a subresource that sets one of these fields would make all updates permafail

I'll enforce on subresource.  In the end, probably.  You good with a TODO?